File: annotations/src/main/java/com/bennyhuo/tieguanyin/annotations/Optional.java
package com.bennyhuo.tieguanyin.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by benny on 1/29/18.
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.CLASS)
public @interface Optional {
    @Deprecated
    String stringValue() default "";
    @Deprecated
    char charValue() default '0';
    @Deprecated
    byte byteValue() default 0;
    @Deprecated
    short shortValue() default 0;
    @Deprecated
    int intValue() default 0;
    @Deprecated
    long longValue() default 0;
    @Deprecated
    float floatValue() default 0f;
    @Deprecated
    double doubleValue() default 0.0;
    @Deprecated
    boolean booleanValue() default false;
}

File: annotations/src/main/java/com/bennyhuo/tieguanyin/annotations/Optional.kt
package com.bennyhuo.tieguanyin.annotations

/**
 * Created by benny on 1/29/18.
 */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.BINARY)
annotation class Optional(
    val stringValue: String = "",
    val charValue: Char = '0',
    val byteValue: Byte = 0,
    val shortValue: Short = 0,
    val intValue: Int = 0,
    val longValue: Long = 0,
    val floatValue: Float = 0f,
    val doubleValue: Double = 0.0,
    val booleanValue: Boolean = false
)